src/components/vm/disks/diskAdd.jsx

```diff
@@ -513,11 +514,49 @@ export class AddDiskModalBody extends React.Component {
             break;
         }
         case 'file': {
+            this.setState({ file: value });
+
             if (value.endsWith(".iso")) {
                 // use onValueChange instead of setState in order to perform subsequent state change logic
                 this.onValueChanged("device", "cdrom");
             }
-            this.setState({ file: value });
+
+            if (value) {
+                cockpit.spawn(["/bin/bash", "-c", "xargs -0n1 head --bytes=16 "], { binary: true, err: "message", superuser: "try" })
```

Comment:
Why does this need a shell wrapper? I agree that we can't use cockpit.file() for this, but calling just head directly ought to be fine.

You can also eliminate the xargs by just calling `["head", "--bytes=16", value]`, which is much simpler.